tests/disk/disk.go

```diff
@@ -0,0 +1,197 @@
+	"strings"
+	"time"
+)
+
+// Disk has the attributes of a virtual disk which is emulated for integration
+// testing.
+type Disk struct {
+	// Size in bytes
+	Size int64
+	// The backing image name
+	// eg: /tmp/fake123
+	imageName string
+	// the disk name
+	// eg: /dev/loop9002
+	DiskPath string
+	// mount point if any
+	MountPoints []string
+}
+
+// NewDisk creates a Disk struct, with a specified size. Also the
+// random disk image name is generated. The actual image is generated only when
+// we try to attach the disk
+func NewDisk(size int64) Disk {
+	disk := Disk{
+		Size:      size,
+		imageName: generateDiskImageName(),
+		DiskPath:  "",
+	}
+	return disk
+}
+
+// Generates a random image name for the backing file.
+// the file name will be of the format fakeXXX, where X=[0-9]
+func generateDiskImageName() string {
+	rand.Seed(time.Now().UTC().UnixNano())
+	randomNumber := 100 + rand.Intn(899)
+	imageName := "fake" + strconv.Itoa(randomNumber)
+	return os.TempDir() + "/" + imageName
+}
+
+func (disk *Disk) createDiskImage() error {
+	f, err := os.Create(disk.imageName)
+	if err != nil {
+		return fmt.Errorf("error creating disk image. Error : %v", err)
+	}
+	err = f.Truncate(disk.Size)
+	if err != nil {
+		return fmt.Errorf("error truncating disk image. Error : %v", err)
+	}
+
+	return nil
+}
+
+// CreateLoopDevice creates a loopback device if the disk is not present
+func (disk *Disk) CreateLoopDevice() error {
+	if disk.DiskPath == "" {
+		var err error
+		if _, err = os.Stat(disk.imageName); err != nil && os.IsNotExist(err) {
+			err = disk.createDiskImage()
+			if err != nil {
+				return err
+			}
+		} else if err != nil {
+			return err
+		}
+		deviceName := getLoopDevName()
+		devicePath := "/dev/" + deviceName
+		// create the loop device using losetup
+		createLoopDeviceCommand := "losetup " + devicePath + " " + disk.imageName + " --show"
+		err = RunCommandWithSudo(createLoopDeviceCommand)
+		if err != nil {
+			return fmt.Errorf("error creating loop device. Error : %v", err)
+		}
+		disk.DiskPath = devicePath
+	}
+	return nil
+}
+
+// Generates a random loop device name. The name will be of the
+// format loop9XXX, where X=[0-9]. The 4 digit numbering is chosen so that
+// we get enough disks to be randomly generated and also it does not clash
+// with the existing loop devices present in some systems.
+func getLoopDevName() string {
+	rand.Seed(time.Now().UTC().UnixNano())
+	randomNumber := 9000 + rand.Intn(999)
+	diskName := "loop" + strconv.Itoa(randomNumber)
+	return diskName
+}
+
+// RunCommandWithSudo runs a command with sudo permissions
+func RunCommandWithSudo(cmd string) error {
+	return RunCommand("sudo " + cmd)
+}
+
+// RunCommand runs a command on the host
+func RunCommand(cmd string) error {
+	substring := strings.Fields(cmd)
+	name := substring[0]
+	args := substring[1:]
+	stdout, err := exec.Command(name, args...).CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("run failed %s %v %v", cmd, err, stdout)
```

Comment:
To identify the value in logging
```suggestion
		return fmt.Errorf("run failed, cmd={%s} error={%v} output={%v}", cmd, err, stdout)
```